src/main/java/seedu/address/logic/commands/AddAssignmentCommand.java

```diff
@@ -18,30 +26,42 @@ public class AddAssignmentCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an Assignment to the app. "
             + "Parameters: "
+            + PREFIX_STUDENT_INDEX + "STUDENT INDEX "
             + PREFIX_ASSIGNMENT_NAME + "ASSIGNMENT NAME "
             + PREFIX_ASSIGNMENT_MAX_SCORE + "MAX SCORE "
             + "Example: " + COMMAND_WORD + " "
+            + PREFIX_STUDENT_INDEX + "1"
             + PREFIX_ASSIGNMENT_NAME + "Assignment 1"
             + PREFIX_ASSIGNMENT_MAX_SCORE + "100";
     public static final String MESSAGE_SUCCESS = "New assignment added: %1$s";
     public static final String MESSAGE_DUPLICATE_ASSIGNMENT = "This assignment already exists in the address book";
-    private final Assignment toAdd;
+    private final AssignmentDescriptor toAddDescriptor;
+    private final Index index;
 
     /**
      * Creates an AddAssignmentCommand to add the specified {@code Assignment}
      */
-    public AddAssignmentCommand(Assignment assignment) {
-        requireNonNull(assignment);
-        toAdd = assignment;
+    public AddAssignmentCommand(Index index, AssignmentDescriptor toAddDescriptor) {
+        requireNonNull(toAddDescriptor);
+        requireNonNull(index);
+        this.index = index;
+        this.toAddDescriptor = toAddDescriptor;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        if (model.hasAssignment(toAdd)) {
+        List<Student> lastShownList = model.getFilteredStudentList();
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+        Student studentToAddAssignmentTo = lastShownList.get(index.getZeroBased());
+        Assignment toAdd = createAssignment(studentToAddAssignmentTo, toAddDescriptor);
+        if (studentToAddAssignmentTo.hasAssignment(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_ASSIGNMENT);
         }
-        model.addAssignment(toAdd);
+        Student edittedStudent = studentToAddAssignmentTo.addAssignment(toAdd);
+        model.setStudent(studentToAddAssignmentTo, edittedStudent);
```

Comment:
Should rename "edittedStudent" to "editedStudent". Mispelling of "edited".

Reply:
okay, fixed!